### ISSUE TITLE
scss-lint enexpected error on exit:  rb_xmalloc_mul_add cannot be found

### ISSUE THREAD

**Issue by gogainda** [oracle/truffleruby#2151]
For **truffleruby 21.0.0-dev-0b6d7ffe, like ruby 2.7.2, GraalVM CE Native [x86_64-darwin]**
4 failures(expected) and some unexpected error on exit:
```
bundler: failed to load command: rspec (/Users/novoi/.rubies/truffleruby-dev/bin/rspec)
RuntimeError: External LLVMFunction rb_xmalloc_mul_add cannot be found. (LLVMLinkerException)
Translated to internal error
  node.c:1187:in `ast_newnode_in_bucket'
  node.c:1214:in `rb_ast_newnode'
  ripper.y:9527:in `node_newnode'
  ripper.y:555:in `ripper_new_yylval'
  ripper.y:5463:in `ripper_yylval_id'
  ripper.y:8738:in `parse_ident'
  ripper.y:9497:in `parser_yylex'
  ripper.y:9507:in `yylex'
  ripper.c:5819:in `ripper_yyparse'
  ripper.y:12991:in `ripper_parse0'
  /Users/novoi/.rubies/truffleruby-dev/lib/truffle/truffle/cext.rb:1464:in `rb_ensure'
  exception.c:68:in `rb_ensure'
  ripper.y:13029:in `ripper_parse'
  /Users/novoi/.rubies/truffleruby-dev/lib/truffle/truffle/cext_ruby.rb:39:in `parse'
  /Users/novoi/.rubies/truffleruby-dev/lib/mri/ripper/sexp.rb:33:in `sexp'
  /Users/novoi/.rubies/truffleruby-dev/lib/gems/gems/rspec-support-3.10.0/lib/rspec/support/source.rb:55:in `ast'
  /Users/novoi/.rubies/truffleruby-dev/lib/gems/gems/rspec-support-3.10.0/lib/rspec/support/source.rb:71:in `nodes_by_line_number'
  /Users/novoi/.rubies/truffleruby-dev/lib/gems/gems/rspec-core-3.10.0/lib/rspec/core/formatters/snippet_extractor.rb:118:in `location_nodes_at_beginning_line'
  /Users/novoi/.rubies/truffleruby-dev/lib/gems/gems/rspec-core-3.10.0/lib/rspec/core/formatters/snippet_extractor.rb:96:in `expression_node'
  /Users/novoi/.rubies/truffleruby-dev/lib/gems/gems/rspec-core-3.10.0/lib/rspec/core/formatters/snippet_extractor.rb:88:in `line_range_of_location_nodes_in_expression'
  /Users/novoi/.rubies/truffleruby-dev/lib/gems/gems/rspec-core-3.10.0/lib/rspec/core/formatters/snippet_extractor.rb:57:in `line_range_of_expression'
  /Users/novoi/.rubies/truffleruby-dev/lib/gems/gems/rspec-core-3.10.0/lib/rspec/core/formatters/snippet_extractor.rb:42:in `expression_lines'
  /Users/novoi/.rubies/truffleruby-dev/lib/gems/gems/rspec-core-3.10.0/lib/rspec/core/formatters/snippet_extractor.rb:31:in `extract_expression_lines_at'
  /Users/novoi/.rubies/truffleruby-dev/lib/gems/gems/rspec-core-3.10.0/lib/rspec/core/formatters/exception_presenter.rb:231:in `read_failed_lines'
  /Users/novoi/.rubies/truffleruby-dev/lib/gems/gems/rspec-core-3.10.0/lib/rspec/core/formatters/exception_presenter.rb:166:in `failure_slash_error_lines'
  /Users/novoi/.rubies/truffleruby-dev/lib/gems/gems/rspec-core-3.10.0/lib/rspec/core/formatters/exception_presenter.rb:153:in `block in failure_lines'
  <internal:core> core/kernel.rb:533:in `tap'
  /Users/novoi/.rubies/truffleruby-dev/lib/gems/gems/rspec-core-3.10.0/lib/rspec/core/formatters/exception_presenter.rb:152:in `failure_lines'
  /Users/novoi/.rubies/truffleruby-dev/lib/gems/gems/rspec-core-3.10.0/lib/rspec/core/formatters/exception_presenter.rb:34:in `colorized_message_lines'
  /Users/novoi/.rubies/truffleruby-dev/lib/gems/gems/rspec-core-3.10.0/lib/rspec/core/formatters/exception_presenter.rb:253:in `formatted_message_and_backtrace'
  /Users/novoi/.rubies/truffleruby-dev/lib/gems/gems/rspec-core-3.10.0/lib/rspec/core/formatters/exception_presenter.rb:88:in `fully_formatted_lines'
  /Users/novoi/.rubies/truffleruby-dev/lib/gems/gems/rspec-core-3.10.0/lib/rspec/core/formatters/exception_presenter.rb:80:in `fully_formatted'
  /Users/novoi/.rubies/truffleruby-dev/lib/gems/gems/rspec-core-3.10.0/lib/rspec/core/notifications.rb:200:in `fully_formatted'
  /Users/novoi/.rubies/truffleruby-dev/lib/gems/gems/rspec-core-3.10.0/lib/rspec/core/notifications.rb:114:in `block in fully_formatted_failed_examples'
  <internal:core> core/enumerable.rb:353:in `each_with_index'
  /Users/novoi/.rubies/truffleruby-dev/lib/gems/gems/rspec-core-3.10.0/lib/rspec/core/notifications.rb:113:in `fully_formatted_failed_examples'
  /Users/novoi/.rubies/truffleruby-dev/lib/gems/gems/rspec-core-3.10.0/lib/rspec/core/formatters/base_text_formatter.rb:32:in `dump_failures'
  /Users/novoi/.rubies/truffleruby-dev/lib/gems/gems/rspec-core-3.10.0/lib/rspec/core/reporter.rb:209:in `block in notify'
  /Users/novoi/.rubies/truffleruby-dev/lib/gems/gems/rspec-core-3.10.0/lib/rspec/core/reporter.rb:208:in `each'
  /Users/novoi/.rubies/truffleruby-dev/lib/gems/gems/rspec-core-3.10.0/lib/rspec/core/reporter.rb:208:in `notify'
  /Users/novoi/.rubies/truffleruby-dev/lib/gems/gems/rspec-core-3.10.0/lib/rspec/core/reporter.rb:178:in `block in finish'
  /Users/novoi/.rubies/truffleruby-dev/lib/gems/gems/rspec-core-3.10.0/lib/rspec/core/reporter.rb:194:in `close_after'
  /Users/novoi/.rubies/truffleruby-dev/lib/gems/gems/rspec-core-3.10.0/lib/rspec/core/reporter.rb:174:in `finish'
  /Users/novoi/.rubies/truffleruby-dev/lib/gems/gems/rspec-core-3.10.0/lib/rspec/core/reporter.rb:76:in `report'
  /Users/novoi/.rubies/truffleruby-dev/lib/gems/gems/rspec-core-3.10.0/lib/rspec/core/runner.rb:115:in `run_specs'
  /Users/novoi/.rubies/truffleruby-dev/lib/gems/gems/rspec-core-3.10.0/lib/rspec/core/runner.rb:89:in `run'
  /Users/novoi/.rubies/truffleruby-dev/lib/gems/gems/rspec-core-3.10.0/lib/rspec/core/runner.rb:71:in `run'
  /Users/novoi/.rubies/truffleruby-dev/lib/gems/gems/rspec-core-3.10.0/lib/rspec/core/runner.rb:45:in `invoke'
  /Users/novoi/.rubies/truffleruby-dev/lib/gems/gems/rspec-core-3.10.0/exe/rspec:4:in `<top (required)>'
  <internal:core> core/kernel.rb:401:in `load'
  <internal:core> core/kernel.rb:401:in `load'
  /Users/novoi/.rubies/truffleruby-dev/bin/rspec:23:in `<top (required)>'
  <internal:core> core/kernel.rb:401:in `load'
  <internal:core> core/kernel.rb:401:in `load'
```
For **truffleruby 21.0.0-dev-f8b272f9, like ruby 2.6.6, GraalVM CE Native [x86_64-darwin]** release
it is still 4 failures but exception details are correct:
```
Failures:

  1) SCSSLint::Linter::TrailingSemicolon when triggering a bug based on Windows IBM437 encoding 
     Failure/Error: @tree = @engine.to_tree
     
     Sass::SyntaxError:
       Invalid CSS after "  content: '▼'": expected ";", was ";"

```

**Comment by eregon** [comment]
Thanks for the report, we need to implement `rb_xmalloc_mul_add`, I'll do that.

**Comment by gogainda** [comment]
strange that it didn't happen in older builds

**Comment by eregon** [comment]
We just upgraded to 2.7, and that uses a newer version of Ripper, which uses that function.
